https://issues.redhat.com/browse/ACM-15420

dockerymick opened the pull request.

dockerymick committed:
https://issues.redhat.com/browse/ACM-15420

oafischer commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
+  workers: <3> 
 ----
+<1> Define the global settings for all managed clusters that have the observability add-on enabled.
+<2> Used to indicate if the observability add-on is enabled to push metrics to hub cluster server.
```
If I remember correctly we aren't supposed to use callouts for additional information, they are only for variable replacements.

dockerymick replied:
@oafischer The additional information here is used to explain why the parameters are used. Since last release, I have used callouts as a solution to remove hashtags out of YAMLs.

oafischer replied:
I'm OK with the suggestions you provide further down but we might need to bring this up in a meeting since I've been tasked with removing callouts that aren't about replacing variables in the past. Thanks for finding a compromise in the meantime. I'll approve after merging the suggestions.

dockerymick replied:
@oafischer ok cool, we can discuss in the team channel before I merge.

swopebe replied:
This is I think not something we need to stress over, as I see teams use these callouts for a number of things now that I look at it. I don't agree with overusage, or even some of the writing here--or using two at one time for the same line, but see that they are used here by the telco/ocp writer:

https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/edge_computing/ztp-preparing-the-hub-cluster#ztp-preparing-the-ztp-git-repository_ztp-preparing-the-hub-cluster

swopebe replied:
Maybe when we first started using them, because we didn't before, we had guidance to only use them for variables but I lean towards using them to replace any #comments-that-clutter-the-code.

swopebe replied:
See here where the same product doc isn't clean and the code is cluttered with comments:

https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/networking/networking-operators#viewing-stats-collected-kubernetes-nmtate-op_k8s-nmstate-operator

swopebe replied:
But I will say that since you have punctuation, these need to be full sentences. 

Used to indicate if the observability add-on is enabled to push metrics to hub cluster server. (not complete)

Also, we agreed that unless in code, Observability is proper. You can do either way here. :-)

dockerymick replied:
@swopebe thanks for your comments and examples. I have updated the file with complete sentence and capitalized "observability" where needed. Awaiting review from @saswatamcode

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
+    enableMetrics: false <2>
+  workers: <3> 
 ----
+<1> Define the global settings for all managed clusters that have the observability add-on enabled.
```
```suggestion
<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the observability add-on enabled.
```

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
+  workers: <3> 
 ----
+<1> Define the global settings for all managed clusters that have the observability add-on enabled.
+<2> Used to indicate if the observability add-on is enabled to push metrics to hub cluster server.
```
```suggestion
<2> Use the `enableMetrics` parameter to indicate that the observability add-on is enabled to push metrics to hub cluster server.
```

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
 ----
+<1> Define the global settings for all managed clusters that have the observability add-on enabled.
+<2> Used to indicate if the observability add-on is enabled to push metrics to hub cluster server.
+<3> Used to list worker nodes into the metric collector process to shard federate requests made to Prometheus on your hub cluster. Then the metric collector sends sperate remote-write requests to Thanos on your hub cluster.  
```
```suggestion
<3> Use the `workers` parameter to list worker nodes into the metric collector process to shard federate requests made to Prometheus on your hub cluster. Then the metric collector sends sperate remote-write requests to Thanos on your hub cluster.  
```

oafischer committed:
https://issues.redhat.com/browse/ACM-15420

Co-authored-by: Mikela Jackson <mdockery@redhat.com>

oafischer committed:
https://issues.redhat.com/browse/ACM-15420

Co-authored-by: Mikela Jackson <mdockery@redhat.com>

oafischer committed:
https://issues.redhat.com/browse/ACM-15420

Co-authored-by: Mikela Jackson <mdockery@redhat.com>

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
+    enableMetrics: false <2>
+  workers: <3> 
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the observability add-on enabled.
```
```suggestion
<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
```

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -574,9 +574,13 @@ Update the `multicluster-observability-operator` resource by setting `enableMetr
+  workers: <3> 
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the observability add-on is enabled to push metrics to hub cluster server.
```
```suggestion
<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
```

dockerymick committed:
Apply suggestions from code review

dockerymick committed:
Update observability_enable.adoc

Updating the capitalization for Observability and replaced ROSA reference with the conref.

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
   imagePullSecret: multiclusterhub-operator-pull-secret
-  observabilityAddonSpec: # The ObservabilityAddonSpec defines the global settings for all managed clusters which have observability add-on enabled
-    enableMetrics: false #indicates the observability addon push metrics to hub server
+  observabilityAddonSpec: <1>
+    enableMetrics: false <2>
+  workers: <3> 
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> Use the `workers` parameter to list worker nodes into the metric collector process to shard federate requests made to Prometheus on your hub cluster. Then the metric collector sends sperate remote-write requests to Thanos on your hub cluster.  
```
@saswatamcode when you have a chance, will you review this part of the PR please?

saswatamcode replied:
Hi @dockerymick! We are dealing with an issue at the moment regarding this exact feature, which is why I'm holding off on commenting.

But I think we can change it a bit in 2.12.2. For now I would recommend the following,
```suggestion
  observabilityAddonSpec: <1>
    enableMetrics: false <2>
  workers: <3> 
----
<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. These workers will shard federate requests made to Prometheus on your hub cluster, amongst themselves, and send seperate remote-write requests to Thanos on your hub cluster.  
```

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> Use the `workers` parameter to list worker nodes into the metric collector process to shard federate requests made to Prometheus on your hub cluster. Then the metric collector sends sperate remote-write requests to Thanos on your hub cluster.  
```
@saswatamcode thanks for your suggestion, here is a similar suggestion based on the IBM Style Guide (Anthropomorphism, and [maintaining present/active tone](https://www.ibm.com/docs/en/ibm-style?topic=grammar-verbs).
```suggestion
<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard federate requests made to Prometheus on your hub cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```

dockerymick committed:
Update observability/observability_enable.adoc

dockerymick commented:
@oafischer @saswatamcode thanks for yalls reviews. I made a change based on @saswatamcode suggestion. When you both have an opportunity, please re-review this PR

oafischer commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard federate requests made to Prometheus on your hub cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```
Is this supposed to be "share separate?"
```suggestion
<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers share separate requests made to Prometheus on your hub cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```

saswatamcode replied:
Nope it is shard federate :)

saswatamcode replied:
Maybe /federate to notify it is an endpoint

saswatamcode replied:
@dockerymick we need to also remove the "on the hub cluster" after Prometheus
```suggestion
<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard /federate requests made to Prometheus on spoke clusters, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```

dockerymick replied:
If it is an endpoint, i suggest that it be monospaced. We also do not use "spoke cluster" in the doc. It is managed cluster. 
```suggestion
<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```

dockerymick committed:
Update observability/observability_enable.adoc

dockerymick commented:
@oafischer @saswatamcode commit made with some adjustments

jacobbaungard commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```
As far as I understand this feature is in "Tech Preview". Do we want to make a note of this? cc: @saswatamcode

oafischer replied:
Tech preview does need to be noted in the docs. @dockerymick

dockerymick replied:
Thanks for the reminder! @oafischer @jacobbaungard I just made updates related to the reminder

jacobbaungard reviewed (approved):
Looks fine to me but perhaps @saswatamcode who worked on this feature would want to have a look as well?

dockerymick commented on release_notes/acm_whats_new.adoc:
```diff
@@ -84,6 +84,8 @@ For other Application topics, see link:../applications/app_management_overview.a
 
 * You can now use the _Advanced search_ option from the console by selecting the *Advanced search* drop-down button. Specify your query and receive results that match the exact strings that you enter and range-based search parameters. See link:../console/search_console.adoc#search-customization[Search customization and configurations].
 
+* Use the new `workers` parameter in the `ObservabilityAddOn` custom resource definition to add more worker nodes into the metric collector procress to shard federate requests made to your hub cluster. See link:../observability/observability_enable.adoc#enabling-observability-service[Enabling the observability service].
```
```suggestion
* *Technology Preview*: Use the new `workers` parameter in the `ObservabilityAddOn` custom resource definition to add more worker nodes into the metric collector procress to shard federate requests made to your hub cluster. See link:../observability/observability_enable.adoc#enabling-observability-service[Enabling the observability service].
```

dockerymick committed:
Update release_notes/acm_whats_new.adoc

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```
```suggestion
<3> *Technology Preview*: Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```

dockerymick committed:
Update observability/observability_enable.adoc

saswatamcode commented on observability/observability_enable.adoc:
```diff
@@ -559,40 +559,45 @@ thanos --version
 ----
+<1> Use the `observabilityAddonSpec` parameter to define the global settings for all managed clusters that have the Observability add-on enabled.
+<2> Use the `enableMetrics` parameter to indicate that the Observability add-on is enabled to push metrics to hub cluster server.
+<3> *Technology Preview*: Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster, to the internal workers, and sends separate remote-write requests to Thanos on your hub cluster.  
```
Ok as of 2.12.2 this will change a bit, I merged the changes just today.
But for now, final suggestions,
```suggestion
<3> *Technology Preview*: Use the `workers` parameter to increase the number of internal workers in the metric collector process. The internal workers shard `/federate` endpoint requests made to Prometheus on your managed cluster and send separate remote-write requests to Thanos on your hub cluster.  
```

dockerymick committed:
Update observability/observability_enable.adoc

Co-authored-by: Saswata Mukherjee <saswataminsta@yahoo.com>

dockerymick commented:
We agreed that there should be a new issue opened to address the changes related to 2.12.2. My recommendation made through Slack to @saswatamcode: 

"based on the z-stream excel sheet, It looks like ACM 2.12.2 will be released [in the new year]. I recommend that a new issue be created and the change be addressed in another PR. I will be on leave [soon], so having the issue will be helpful to Oliver, who is helping with the Observability docs (in case I do not get to the update before I leave). Considering the holidays and recharge coming up, I’m not sure that the update will get in before the end of this week. But I should be able to get to it in the New Year."

From my perspective, it seems like we can move forward and merge this PR.

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -21,7 +21,7 @@ Monitor the status of your managed clusters with the observability component, al
 - Direct network access to the hub cluster is required. Network access to load balancers and proxies are not supported. For more information, see link:../networking/networking_intro.adoc#networking[Networking].
 - You must configure an object store to create a storage solution. 
-**  *Important:* When you configure your object store, ensure that you meet the encryption requirements that are necessary when sensitive data is persisted. The observability service uses Thanos supported, stable object stores. You might not be able to share an object store bucket by multiple {acm-short} observability installations. Therefore, for each installation, provide a separate object stor ...
+**  *Important:* When you configure your object store, ensure that you meet the encryption requirements that are necessary when sensitive data is persisted. The Observability service uses Thanos supported, stable object stores. You might not be able to share an object store bucket by multiple {acm-short} Observability installations. Therefore, for each installation, provide a separate object stor ...
```
Removing extra space
```suggestion
** *Important:* When you configure your object store, ensure that you meet the encryption requirements that are necessary when sensitive data is persisted. The Observability service uses Thanos supported, stable object stores. You might not be able to share an object store bucket by multiple {acm-short} Observability installations. Therefore, for each installation, provide a separate object store bucket. 
```

dockerymick committed:
Update observability/observability_enable.adoc
